src/daemon/mod.rs

```diff
@@ -695,14 +694,170 @@ fn create_password(prompt: &str) -> std::io::Result<String> {
         .interact_on(&term)
 }
 
+pub async fn open_forest_car_union_db(
```

Comment:
I don't understand what this function is supposed to do. It appears that it does several unrelated things (like opening the current database AND handling snapshot importing). Could you write some documentation for it?

Reply:
In the current implementation, to avoid creating a wrapper type of `Mutex<ManyCar>` that does not require `&mut self` for APIs like `fn read_only`, snapshot import is moved to before DB initialization, thus any functions that require a mutable `ManyCar` is moved into this function(snapshot import, auto download snapshot).

Reply:
Doc added